### PR TITLE
Add ability to filter contracts of different statuses for hasContract

### PR DESCRIPTION
**Pull request by vonElfvin**
Hide whitespace changes 😇

### CODE REVIEW

File: product-pricing-service/src/main/java/com/hedvig/productPricing/campaign/util/DateUtil.kt [hunk 1]
```diff
@@ -9,50 +9,54 @@ import java.time.ZoneId
 import java.time.ZoneOffset
 
 class DateUtil {
-  companion object {
```
Comment (FredrikAreschoug) [inline product-pricing-service/src/main/java/com/hedvig/productPricing/campaign/util/DateUtil.kt:12]: I know this it really not a part of your PR but why is this a class and then everything is in a companion object couldn't a `DateUtil` just be an object?

File: product-pricing-service/src/main/java/com/hedvig/contract/web/InternalContractController.kt [hunk 2]
```diff
@@ -31,125 +32,128 @@ import java.util.UUID
...
-    )
-  }
 
-  @GetMapping("/members/{memberId}/hasContract")
-  fun hasContract(@PathVariable memberId: String): ResponseEntity<Boolean> =
-    ResponseEntity.ok(contractService.hasContract(memberId))
+    @GetMapping("/members/{memberId}/hasContract")
+    fun hasContract(
+        @PathVariable memberId: String,
+        @RequestParam(required = false) statusFilter: ContractStatusFilter = ContractStatusFilter.ANY
```
Comment (FredrikAreschoug) [inline product-pricing-service/src/main/java/com/hedvig/contract/web/InternalContractController.kt:152]: I like it!

**Review by FredrikAreschoug — COMMENTED** [review]
I like it my only suggestion is that It should really be `ContractStateFilter` and not status. But since But `ContractStatus` is the one that is wrong in the first place. So either or

File: product-pricing-service/src/main/java/com/hedvig/productPricing/campaign/util/DateUtil.kt (hunk 1 above)
Comment (vonElfvin) [reply product-pricing-service/src/main/java/com/hedvig/productPricing/campaign/util/DateUtil.kt]: Good catch, I changed it :)

File: product-pricing-service/src/main/java/com/hedvig/contract/web/InternalContractController.kt (hunk 2 above)
Comment (vonElfvin) [reply product-pricing-service/src/main/java/com/hedvig/contract/web/InternalContractController.kt:152]: Awesome!

**Review by FredrikAreschoug — APPROVED** [review]
:tada:

**Pull request merged**